pyam_analysis/core.py

```diff
@@ -536,6 +537,30 @@ def metadata(self, meta=None, name=None, filters={},
                 meta = meta[keep_col_match(meta[col], values)]
             return return_df(meta, display, idx_cols)
 
+    def to_excel(self, excel_writer, filters={}, exclude_cat=[],
+                 sheet_name='data', index=False, **kwargs):
+        """Write timeseries data to Excel using the IAMC template convention
+        (wrapper for `pandas.DataFrame.to_excel()`)
+
+        Parameters
+        ----------
+        excel_writer: string or ExcelWriter object
+             file path or existing ExcelWriter
+        filters: dict, optional
+            filter by model, scenario, region, variable, level, year, category
+            see function _select() for details
+        exclude_cat: None or list of strings, default []
+            exclude all scenarios from the listed categories
+        sheet_name: string, default 'data'
+            name of the sheet that will contain the (filtered) IamDataFrame
+        index: boolean, default False
+            write row names (index)
+        """
+        df = self.timeseries(filters, exclude_cat).reset_index()
```

Comment:
not sure if you want to think about how this will work in the future, the internal data representation is long-form (rather than wide-form). I might think that the call we want to make is as follows:

```
(df.filter(filters=foo)
    .add_metadata() # I think I added this function, but users can optionally do it here
    .to_wide()
    .capitalize_columns() # there might be a better pandas function here to call directly?
    .to_excel(*args, **kwargs) # can just pass them directly to pandas at this point
)
```

Reply:
agree on using the `filter()` function, but disagree on adding the meta-data - the idea is to get something that conforms 100% to the "old" template, so that it can be used by the current web databases or the `Riamc` library...

Also, adding (let's say) 10 meta-data entries for a scenario with thousands of variables and ~10 timesteps roughly doubles the resulting file size.

Reply:
re long vs. wide - that's what I have been (ab)using the `timeseries()`` function for. My hunch is that long is more efficient if we have timeseries data that is sparse in years...

Reply:
re capitalize columns - having both strings and ints in the column header is causing problems for shorter notation...